### ISSUE TITLE
CloudFoundry + MongoDB is broken

### ISSUE THREAD

**Issue by jdubois** [jhipster/generator-jhipster#4363]
From my stats, very few people are using CloudFoundry (10 times less than Heroku!), which is probably why we didn't see this.
Besides, there is only a "free" plan, which is very limited and not production-ready, on the CloudFoundry marketplace.

Choosing CloudFoundry + MongoDB doesn't work as we have both `CloudDatabaseConfiguration` and `CloudMongoDbConfiguration` which get configured, and which configure the same bean, so Spring crashes at startup. This just cannot work!!

It also seems Spring Cloud has changed its MongoDB support, so the `CloudMongoDbConfiguration` doesn't seem to work anymore: when using it, I have connection exceptions to the MongoDB database, so it's clear that the MongoDB configuration isn't automatically setup from Spring Cloud.

Of course it works if you code directly database credentials in the `application-prod.yml`.

I'm also pinging @jkutner as there might be the same kind of issue on Heroku, as it's very similar.

**Comment by jkutner** [comment]
No problem on Heroku (from what I can see with a smoke test of 3.9.1). Please let me know if there is something more subtle.

**Comment by jdubois** [comment]
I will commit a fix soon

**Comment by jdubois** [comment]
I will merge this with #4349 as it's related

**Comment by ehillis** [comment]
I made a comment on this "old issue" because I didn't see it resolved anywhere else..   is the circular reference fixed (or about to be?)    I'm using bluemix because it offers a free (30 day trial), and I assumed that to use heroku with mongodb I would need to pay $18/month just to "try" it...  heroku did NOT seem to work for me (out of the box), it was getting an error with a MONGODB environmental variable or some such.. so I assumed I needed to add the Mongo DB Compose or whatever..    if I don't need to do that, would appreciate help on this also.. don't have any attachment to bluemix (other than someone offers to pay for it and it give 30 days free).

**Comment by jdubois** [comment]
@ehillis it should work now on Cloud Foundry
There shouldn't be such a problem on Heroku, as it's not working the same way (or this is a whole different issue)

**Comment by jdubois** [comment]
Yes I'm going to fix this for CloudFoundry, and test if it still works for
Heroku.
I have everything working, but I need time to test. And I have no idea when
I will have some free time.

Le 21 oct. 2016 8:33 AM, "ehillis" <notifications@github.com> a écrit :

I made a comment on this "old issue" because I didn't see it resolved
anywhere else.. is the circular reference fixed (or about to be?) I'm using
bluemix because it offers a free (30 day trial), and I assumed that to use
heroku with mongodb I would need to pay $18/month just to "try" it...
heroku did NOT seem to work for me (out of the box), it was getting an
error with a MONGODB environmental variable or some such.. so I assumed I
needed to add the Mongo DB Compose or whatever.. if I don't need to do
that, would appreciate help on this also.. don't have any attachment to
bluemix (other than someone offers to pay for it and it give 30 days free).

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/jhipster/generator-jhipster/issues/4363#issuecomment-255305002>,
or mute the thread

**Comment by ehillis** [comment]
Thanks!
Do I need to update Jhipster and recreate my project to get mine to work?

**Comment by jdubois** [comment]
You need to wait for the next release, probably very soon

**Comment by vhiguita** [comment]
Hi, 
My app is deployed in Pivotal web services with the default mlab plan sandbox which it's very limited. I have created a new account with mlab directly with a different plan, is it possible that the deployed app in pivotal could connect directly to this new mlab account?, I modified these lines in the file application-prod.ym and deployed in pivotal again but it's not working.

 mongodb:
       uri: mongodb://<user>:<pwd>@ds137760.mlab.com:37760
       database: database_name

Thank you very much.
![attachment](https://cloud.githubusercontent.com/assets/284188/24295352/5b04f38c-1068-11e7-80a4-1a38ddb68017.jpg)
